From 2d882c60407a9c426667560fbd153315393e1bce Mon Sep 17 00:00:00 2001
From: lijuan <juan.li@spacemit.com>
Date: Fri, 7 Jun 2024 14:14:31 +0800
Subject: [PATCH 0696/1203] k1: rproc: enable rproc module to avoid bus hangs
 dead when cat clk_summary

log err:
[  320.688746] rcu: INFO: Stall ended before state dump start
[  322.957007] rcu: rcu_preempt kthread starved for 2713 jiffies! g-296 f0x0 RCU_GP_WAIT_GPS(1) ->state=0x402 ->cpu=3
[  736.656451] rcu: Unless rcu_preempt kthread gets sufficient CPU time, OOM is now expected behavior.
[  736.656455] rcu: RCU grace-period kthread stack dump:
[  736.656458] task:rcu_preempt     state:I stack:0     pid:16    ppid:2      flags:0x00000000
[  736.656470] Call Trace:
[  736.656474] [<ffffffff80fcced0>] __schedule+0x28e/0x7ae
[  736.686666] rcu: Stack dump where RCU GP kthread last ran:
[  736.692138] Task dump for CPU 3:
[  736.695361] task:swapper/3       state:R  running task     stack:0     pid:0     ppid:1      flags:0x00000000
[  736.705296] Call Trace:
[  736.707732] [<ffffffff80fcced0>] __schedule+0x28e/0x7ae
[  757.669427] rcu: INFO: rcu_preempt detected stalls on CPUs/tasks:
[  757.675543] rcu: 2-...0: (102 ticks this GP) idle=2304/1/0x4000000000000000 softirq=129/131 fqs=2627
[  757.684786] rcu: 5-...0: (33 ticks this GP) idle=270c/1/0x4000000000000000 softirq=27/28 fqs=2628
[  757.693760] rcu: (detected by 4, t=5254 jiffies, g=-295, q=5 ncpus=8)
[  757.700319] Task dump for CPU 2:
[  757.703544] task:cat             state:R  running task     stack:0     pid:163   ppid:158    flags:0x0000000a
[  757.713483] Call Trace:
[  757.715922] [<ffffffff80fcced0>] __schedule+0x28e/0x7ae
[  757.721156] [<ffffffff80fcab96>] do_irq+0x6a/0x7e
[  757.725869] [<ffffffff80fd3e28>] ret_from_exception+0x0/0x64
[  757.731542] [<ffffffff80822abc>] clk_core_is_enabled+0x4/0x9e
[  757.737307] [<ffffffff8082612a>] clk_summary_show+0x88/0xf8
[  757.742888] [<ffffffff80210ada>] seq_read_iter+0xd0/0x31e
[  757.748304] [<ffffffff80210da4>] seq_read+0x7c/0xa8
[  757.753186] [<ffffffff8055002c>] full_proxy_read+0x42/0x68
[  757.758665] [<ffffffff801e7d80>] vfs_read+0x88/0x224
[  757.763644] [<ffffffff801e8700>] ksys_read+0x54/0xcc
[  757.768620] [<ffffffff801e878c>] __riscv_sys_read+0x14/0x1c
[  757.774199] [<ffffffff80fcaa82>] do_trap_ecall_u+0xba/0x12e
[  757.779767] [<ffffffff80fd3e28>] ret_from_exception+0x0/0x64

Signed-off-by: lijuan <juan.li@spacemit.com>
Change-Id: I20a7696f1d53a08a0e234a9593edef856ae31ff4
---
 drivers/remoteproc/k1x-rproc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/remoteproc/k1x-rproc.c b/drivers/remoteproc/k1x-rproc.c
index c92bdb462efa..89a66efa97b3 100644
--- a/drivers/remoteproc/k1x-rproc.c
+++ b/drivers/remoteproc/k1x-rproc.c
@@ -420,8 +420,10 @@ static int spacemit_rproc_probe(struct platform_device *pdev)
 	struct rproc *rproc;
 
 	ret = rproc_of_parse_firmware(dev, 0, &fw_name);
-	if (ret < 0 && ret != -EINVAL)
+	if (ret < 0 && ret != -EINVAL) {
+		dev_err(dev, "failed to get fw_name: %s\n", fw_name);
 		return ret;
+	}
 
 	rproc = devm_rproc_alloc(dev, np->name, &spacemit_rproc_ops,
 				fw_name, sizeof(*priv));
-- 
2.47.0

